### PR TITLE
TECH_DEBT: Updated docs

### PR DESCRIPTION
**Pull request by seanmcilvenna**
### 🛠 Description of Changes

Updating documentation to provide some high-level information about the capabilities of BOTW. A few TODO's in the docs, and more is needed.

### 🧪 Testing Performed

N/A

## Release Notes

- **New Features**
  - Enhanced documentation structure with new sections for Admin UI, Data Acquisition, Measure Evaluation, Normalization, Telemetry, and Tenant/Facility Management.
  - Introduced Swagger UI specifications for API exploration.

- **Documentation**
  - Comprehensive updates across multiple service specifications, detailing API operations, Kafka events, and service functionalities.
  - Added new documentation files for Administrative UI, Data Acquisition, Measure Evaluation, Normalization, Telemetry, Tenant Management, Notification capabilities, and Kafka Retry Topic Pattern, improving clarity and usability.

These updates significantly improve the comprehensiveness and accessibility of our documentation for end-users.

### CODE REVIEW

File: docs/service_specs/tenant.md [hunk 1]
```diff
@@ -46,10 +46,24 @@ The Tenant service is the entry point for configuring a tenant into Link Cloud.
...
+
+The **Tenant** service provides REST endpoints for managing facility configurations and related metadata.
+
+- **GET /api/Facility**: Retrieve a paged list of facilities based on filters such as `facilityId`, `facilityName`, and sorting options.
+- **POST /api/Facility**: Create a new facility configuration.
+- **GET /api/Facility/{facilityId}**: Retrieve a specific facility configuration by `facilityId`.
+- **PUT /api/Facility/{id}**: Update an existing facility configuration by `id`.
+- **DELETE /api/Facility/{facilityId}**: Delete a facility configuration by `facilityId`.
+
+These operations support managing tenant-specific configurations and workflows efficiently.
```
[the remark this answers is not in the document]
Comment (seanmcilvenna) [reply docs/service_specs/tenant.md:69]: I've discussed this with others and they agree that "TECH DEBT" should include documentation changes. We don't want to discourage people from updating documentation at any time.

File: docs/service_specs/census.md [hunk 2]
```diff
@@ -41,10 +41,32 @@ The Census service is primarily responsible for maintaining a tenants admit and
...
+### Census Configuration Management
+
+- **POST /api/census/config**: Creates a new CensusConfig for a facility.
+- **GET /api/census/config/{facilityId}**: Retrieves the CensusConfig for a specific facility.
+- **PUT /api/census/config/{facilityId}**: Updates the CensusConfig for a specific facility.
+- **DELETE /api/census/config/{facilityId}**: Deletes the CensusConfig for a specific facility.
 
-- **Event**: `PatientCensusScheduled`
\ No newline at end of file
+These endpoints enable tenants to manage census data and configurations effectively, ensuring accurate and consistent census management across facilities.
```
[the remark this answers is not in the document]
Comment (seanmcilvenna) [reply docs/service_specs/census.md]: This is documentation about existing functionality, and is not adding any new functionality.

File: docs/functionality/normalization.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+- **Details**:
+    - Identifies source and destination properties using FHIRPath.
+    - Copies the value from the source to the destination.
+- **Use Case**: Populating properties required for measure evaluation or reporting.
+
+---
+
+### 4. Conditional Transform
+
+- **Description**: *(TODO: Document)*
```
[the remark this answers is not in the document]
Comment (seanmcilvenna) [reply docs/functionality/normalization.md:54]: The TODO's will need to be addressed in a follow-up PR.

File: docs/service_specs/bff.md [hunk 4]
```diff
@@ -40,3 +40,29 @@
...
+
+- **POST /api/integration/patient-event**: Produces a patient event for testing purposes.
+- **POST /api/integration/report-scheduled**: Produces a report scheduled event for testing purposes.
+- **POST /api/integration/data-acquisition-requested**: Produces a data acquisition requested event for testing purposes.
+
+### Service Information
+
+- **GET /api/info**: Retrieves basic service information for the BFF service.
+
+These operations enable robust session management, integration testing, and backend connectivity for frontend applications.
```
[the remark this answers is not in the document]
Comment (seanmcilvenna) [reply docs/service_specs/bff.md:68]: Per other discussion, tech debt can include documentation changes.

File: docs/functionality/census_management.md [hunk 5]
```diff
@@ -42,4 +42,8 @@ Currently, Link persists only the **FHIR ID** of each patient. This identifier a
 
 ### Future Considerations
 
-In the interest of enhancing the user interface, Link is considering storing additional data elements, such as the **patient name** associated with each FHIR ID. This would provide users with meaningful patient identifiers, facilitating easier navigation and record management within the Link UI.
\ No newline at end of file
+In the interest of enhancing the user interface, Link is considering storing additional data elements, such as the **patient name** associated with each FHIR ID. This would provide users with meaningful patient identifiers, facilitating easier navigation and record management within the Link UI.---
+
```
[the remark this answers is not in the document]
Comment (seanmcilvenna) [reply docs/functionality/census_management.md:46]: This comment is on an empty line. Disregard.

File: docs/README.md [hunk 6]
```diff
@@ -5,9 +5,17 @@ This page and its references include documentation for Link's services and the f
...
 ## Functionality
 
+* [Admin UI](functionality/admin_ui.md)
 * [Census Management](functionality/census_management.md)
+* [Data Acquisition](functionality/data_acquisition.md)
+* [Measure Evaluation](functionality/measure_eval.md)
+* [Notifications](functionality/notifications.md)
+* [Normalization](functionality/normalization.md)
+* [Kafka Retry Topics](functionality/retry_topics.md)
 * Submission
```
Comment (sdmcgeown) [inline docs/README.md]: Was there supposed to be a newline here between Submission and Folder Structure?

**Review by sdmcgeown — COMMENTED** [review]
I don't think it's a problem introduced in _this_ PR but I noticed that in the [top-level readme.md file](https://github.com/lantanagroup/link-cloud/blob/updated_docs/README.md) there is a broken URL in the table of contents:

[https://github.com/lantanagroup/link-cloud/blob/updated_docs/README.md#link-cloud-services](https://github.com/lantanagroup/link-cloud/blob/updated_docs/README.md#link-cloud-services)
I think, should be pointing to:
[https://github.com/lantanagroup/link-cloud/blob/updated_docs/README.md#documentation](https://github.com/lantanagroup/link-cloud/blob/updated_docs/README.md#documentation)
Or the heading of that section should be updated to `Link Cloud Services` so that the URL works, whichever way is preferred I guess.

File: docs/README.md (hunk 6 above)
Comment (seanmcilvenna) [reply docs/README.md]: Removed.

**Comment by seanmcilvenna** [comment]
> I don't think it's a problem introduced in _this_ PR but I noticed that in the [top-level readme.md file](https://github.com/lantanagroup/link-cloud/blob/updated_docs/README.md) there is a broken URL in the table of contents:
> 
> https://github.com/lantanagroup/link-cloud/blob/updated_docs/README.md#link-cloud-services I think, should be pointing to: https://github.com/lantanagroup/link-cloud/blob/updated_docs/README.md#documentation Or the heading of that section should be updated to `Link Cloud Services` so that the URL works, whichever way is preferred I guess.

Fixed this anyways.

**Review by sdmcgeown — APPROVED** [review]
Looks good

File: docs/functionality/retry_topics.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    - Success: Message is processed and acknowledged
+    - Failure: Message is either:
+        - Scheduled for another retry if within retry limits
+        - Sent to dead-letter queue if retry limits exceeded
+
+## Configuration
+
+### Retry Settings
+
+> TODO: Where is this configuration stored? How is it managed?
```
Comment (dvargaslantana) [inline docs/functionality/retry_topics.md]: The configuration is located in the app config setting for each service:

```
"ConsumerSettings": {
  "ConsumerRetryDuration": [ "PT20S", "PT60S", "PT120S" ],
  "DisableRetryConsumer": false,
  "DisableConsumer": false
}
```

File: docs/functionality/tenant_mgmt.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    - Reporting Periods
+    - Submission Schedules
+
+## Cross-Service Tenant Configuration
+
+A fully functional tenant requires configuration across multiple services. Each service maintains its own tenant-specific configurations using the `facilityId` as the linking identifier.
+
+### Service-Specific Configurations
+
+#### Tenant Service
```
Comment (dvargaslantana) [inline docs/functionality/tenant_mgmt.md:24]: Large changes to the Tenant config API have been made. Some of these bullets can be omitted based on the work done here: https://lantana.atlassian.net/browse/LNK-2738

File: docs/functionality/tenant_mgmt.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+- Monthly Reporting Plans
+    - Report Types
+    - Reporting Periods
+    - Submission Schedules
+
+## Cross-Service Tenant Configuration
+
+A fully functional tenant requires configuration across multiple services. Each service maintains its own tenant-specific configurations using the `facilityId` as the linking identifier.
+
+### Service-Specific Configurations
```
Comment (dvargaslantana) [inline docs/functionality/tenant_mgmt.md:22]: Query Dispatch is missing from this list. It should probably be added until it is officially phased out.

Comment (dvargaslantana) [reply docs/functionality/tenant_mgmt.md:22]: Census service also looks like it's missing

**Pull request merged**